src/libs/relay/conduit_relay_io_blueprint.cpp

```diff
@@ -1825,6 +1751,403 @@ void write_mesh(const Node &mesh,
+        {
+            cycle = dom["state/cycle"].to_int();
+            opts_suffix = "cycle";
+        }
+    }
+
+#ifdef CONDUIT_RELAY_IO_MPI_ENABLED
+    // reduce to get the cycle (some tasks might not have domains)
+    n_local = (int)cycle;
+
+    relay::mpi::min_all_reduce(n_local,
+                               n_reduced,
+                               mpi_comm);
+
+    cycle = n_reduced.as_int();
+
+    // we also need to have all mpi tasks agree on the `opts_suffix`
+    // checking the first mpi task with domains should be sufficient.
+    // find first
+    n_local   = local_num_domains;
+    n_reduced.reset();
+    
+    relay::mpi::all_gather(n_local,
+                           n_reduced,
+                           mpi_comm);
+
+    index_t idx = -1;
+    NodeConstIterator counts_itr = n_reduced.children();
+    for (index_t i = 0; counts_itr.has_next() && idx < 0; i++)
+    {
+        const Node &curr = counts_itr.next();
+        index_t count = curr.to_index_t();
+        if(count > 0)
+        {
+            idx = i;
+        }
+    }
+
+    // now broadcast from idx
+    Node n_opts_suffix;
+    if(par_rank == idx)
+    {
+        n_opts_suffix = opts_suffix;
+    }
+
+    conduit::relay::mpi::broadcast_using_schema(n_opts_suffix,
+                                                idx,
+                                                mpi_comm);
+
+    opts_suffix = n_opts_suffix.as_string();
+
+#endif
+}
+
+//-----------------------------------------------------------------------------
+// make sure some MPI task has data
+//-----------------------------------------------------------------------------
+bool check_mesh_valid_for_save(int &cycle,
+#ifdef CONDUIT_RELAY_IO_MPI_ENABLED
+                               Node &n_local,
+                               Node &n_reduced,
+#endif
+                               Node &multi_dom,
+                               const Node &mesh,
+                               bool &is_valid,
+                               int &par_rank,
+                               int &par_size
+                               CONDUIT_RELAY_COMMUNICATOR_ARG(MPI_Comm mpi_comm))
+{
+#ifdef CONDUIT_RELAY_IO_MPI_ENABLED
+    is_valid = conduit::relay::mpi::io::blueprint::clean_mesh(mesh, multi_dom, mpi_comm);
+#else
+    is_valid = conduit::relay::io::blueprint::clean_mesh(mesh, multi_dom);
+#endif
+
+    par_rank = 0;
+    par_size = 1;
+    // we may not have any domains so init to max
+    cycle = std::numeric_limits<int>::max();
+
+    int local_boolean = is_valid ? 1 : 0;
+    int global_boolean = local_boolean;
+
+
+#ifdef CONDUIT_RELAY_IO_MPI_ENABLED
+    par_rank = relay::mpi::rank(mpi_comm);
+    par_size = relay::mpi::size(mpi_comm);
+
+    // reduce to check to see if any valid data exists
+
+    n_local = (int)cycle;
+    relay::mpi::sum_all_reduce(n_local,
+                               n_reduced,
+                               mpi_comm);
+
+    global_boolean = n_reduced.as_int();
+
```

Comment:
what happens to is_valid, that is only used if it's valid locally vs the return code means global?
Do we need to pass is valid as an out param?